### PR TITLE
[14.0][IMP] sale_product_seasonality: Add product template management

### PR DESCRIPTION
**Pull request by jcoux**
Related to: https://github.com/OCA/product-attribute/pull/871 which add product template on config lines.

### CODE REVIEW

**Review by simahawk — APPROVED** [review]

File: sale_product_seasonality/models/sale_order.py [hunk 1]
```diff
@@ -30,12 +30,19 @@ def _default_seasonal_config_id(self):
...
+                lambda l: l.is_sale_ok(date)
             )
+            for config_line in config_lines:
+                if config_line.product_id:
+                    product_ids.append(config_line.product_id.id)
+                else:
+                    product_ids.extend(
+                        config_line.product_template_id.product_variant_ids.ids
+                    )
+        products = self.env["product.product"].browse(product_ids)
```
Comment (simahawk) [inline sale_product_seasonality/models/sale_order.py]: NOTE: we could probably avoid browsing here as the compute method requires IDS

**Review by simahawk — CHANGES_REQUESTED** [review]
oh, please update tests

File: sale_product_seasonality/models/sale_order.py (hunk 1 above)
Comment (jcoux) [reply sale_product_seasonality/models/sale_order.py]: Fixed with commit amend/push force: https://github.com/OCA/sale-workflow/compare/9a8f2a52369ca025bf1bc3b69c7213da536183c6..6dd2b5faafcc500a546063af005a454b4fcbb170

**Review by simahawk — APPROVED** [review]

**Review by damdam-s — APPROVED** [review]

**Comment by rousseldenis** [comment]
@jcoux @simahawk It seems there is some problems with this module in 14.0 as tests are red.

**Comment by simahawk** [comment]
@rousseldenis yeah, I noticed it, tnx. I'll have a look later or tomorrow

**Comment by simahawk** [comment]
#1641 contains the fix for tests too

**Pull request closed without merging**